torch/distributed/_state_dict_utils.py

```diff
@@ -19,6 +25,16 @@
     from torch.distributed.tensor._utils import compute_local_shape_and_global_offset
 
 
+logger = getLogger()
+logger.setLevel(logging.INFO)
+
+# Allows retrying cudaHostRegister if it fails
+CKPT_PIN_ALLOW_RETRY = os.environ.get("CKPT_PIN_ALLOW_RETRY", "1") == "1"
+# Peeks last cudaError before pinning shared memory
+CKPT_PIN_PEEK_CUDA_ERROR = os.environ.get("CKPT_PIN_PEEK_CUDA_ERROR", "0") == "1"
+# Pops last cudaError before pinning shared memory
+CKPT_PIN_POP_CUDA_ERROR = os.environ.get("CKPT_PIN_POP_CUDA_ERROR", "0") == "1"
```

Comment:
ditto. Should just move to right before they are used and only get the environment variables that are actually used.